src/aria/core/Log.js

```diff
@@ -255,14 +302,26 @@ module.exports = Aria.classDefinition({
         },
 
         /**
-         * Remove all appenders in the logging system
+         * Removes an appender from the logging system
+         * @param {aria.core.log.DefaultAppender} appender An instance of appender to be removed (must implement the
+         * same methods as aria.core.log.DefaultAppender)
+         */
+        removeAppender : function (appender) {
+            this.clearAppenders(appender);
+        },
+
+        /**
+         * Remove all appenders in the logging system, if an appender is passed then will remove just a specific
+         * appender instance
+         * @param {aria.core.log.DefaultAppender} appender An instance of appender to be removed (must implement the
+         * same methods as aria.core.log.DefaultAppender)
          */
-        clearAppenders : function () {
-            var apps = this.getAppenders();
+        clearAppenders : function (appender) {
+            var apps = appender ? [appender] : this.getAppenders().slice(0);
             for (var i = 0, l = apps.length; i < l; i++) {
                 apps[i].$dispose();
+                this._remove(this._appenders, apps[i]);
             }
```

Comment:
By implementing the right loop in `clearAppenders`, we can avoid duplicating the code from `aria.utils.Array`. Just loop over all items (instead of only the item(s) to remove), and if the item is to be removed (either because there is no argument or because the item is the provided argument), remove it from the array with splice and dispose it. We don't need `_indexOf`, `_remove` and `_removeAt`.